### PR TITLE
Introduce Dynaconf

### PR DESCRIPTION
**Pull request by 0snap**
###  :notebook_with_decorative_cover: Description

Switch configuration framework from confuse to [dynaconf](https://github.com/rochacbruno/dynaconf). The PR updates all of Threat Bus, all plugins, all apps, all integration tests and the MISP test utils.

Docs will follow in a separate PR.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/threatbus](https://docs.tenzir.com/threatbus), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

- Craft some config files and Interactively check application startup. Test that all `required` values are present or the validation should complain otherwise.

### CODE REVIEW

**Comment by 0snap** [comment]
@satta ping: we ditch `confuse` in favor of the superior config framework `dynaconf`. This might require re-packaging for Debian.

**Comment by satta** [comment]
> @satta ping: we ditch `confuse` in favor of the superior config framework `dynaconf`. This might require re-packaging for Debian.

Thanks, good to know. For buster, yes. Bullseye will have  `python3-dynaconf (2.2.3-2)`

**Comment by 0snap** [comment]
Hey @satta, I just checked the docs, and we definitely require `dynaconf >= 3.1.X` for two reasons:
1. According to their [changelog](https://github.com/rochacbruno/dynaconf/blob/master/3.x-release-notes.md#breaking-changes) there are breaking changes between v2 and v3, which would affect us. They have deprecated a the usage of a global structure and introduced and object-local settings management. Since we use that (in accordance to the v3 docs), the code is not compatible to dynaconf 2.X.
2. We use dynaconf `Validators` to check the user configurations for correctness. Validators are a new feature as of v3.1

**Comment by satta** [comment]
Got it. This will mean updating and backporting version 3 as well. 
This is pretty similar to the other issue we are facing with pika. Looks like I will need to revisit the question of how to handle multiple versions of the same Python module in Debian.

**Review by lava — COMMENTED** [review]
The code changes itself look good to me, but I failed to pass validation when I tried to start threatbus:
```
$ head -7 config.yaml.example 
logging:
  console: true
  console_verbosity: DEBUG
  file: false
  file_verbosity: DEBUG
  filename: threatbus.log

$ threatbus -c config.yaml.example 
Invalid config: combined validators failed logging.console is required in env main or logging.file is required in env main

$ THREATBUS_LOGGING_CONSOLE=true threatbus -c config.yaml.example 
Invalid config: combined validators failed logging.console is required in env main or logging.file is required in env main

```

It's not completely clear to me what the issue is.

**Comment by 0snap** [comment]
Both things you tried are invalid. You first need to copy the file to something ending in `.yaml` or `.yml`, otherwise dynaconf does not pick it up. So when you follow the readme and `cp config.yaml.example config.yaml` it should work.

Second, as stated in the readme, `THREATBUS_LOGGING_CONSOLE=true` is invalid and should instead have two underscores.
Here's a working example that would make Threat Bus additionally log to a file:
```
THREATBUS_LOGGING__FILE=true threatbus -c config.yaml
```

File: README.md [hunk 1]
```diff
@@ -66,7 +66,21 @@ are more integrations available, so make sure to check out all
...
+
+```sh
+threatbus -c /path/to/your/special-config.yaml
+```
+
+*Environment variables take precedence over config file values. Prefix
+everything with `THREATBUS_`*
+
+```sh
+export THREATBUS_LOGGING__CONSOLE=true
```
[the remark this answers is not in the document]
Comment (0snap) [reply README.md:82]: No, this is correct. See comment below. I added a sentence to the readme to make sure the reader understands it.

**Comment by lava** [comment]
> You first need to copy the file to something ending in .yaml or .yml, otherwise dynaconf does not pick it up.

Shouldn't we then add a validator that checks that the `-c` parameter value ends in `.yaml` or `.yml`?

**Comment by 0snap** [comment]
> Shouldn't we then add a validator that checks that the -c parameter value ends in .yaml or .yml?

While we lose the ability to accept `toml` or otherwise neat config formats, I agree that we not yet documented the alternatives a user can pick and hence added a strict check for all application to exit early for any other file extension than `yaml` or `yml`.

**Review by lava — APPROVED** [review]
The code looks good; I didn't get to do too much local testing but imho it's fine to merge this now and fix any remaining issues as they come up, since it's a month anyways to the next release.

**Pull request merged**